### DISCUSSION TITLE
Support for quic(http/3)

### DISCUSSION THREAD

**Discussion by hesamhamdarsi** [Kong/kong#8983] [Ideas and feature requests]
Hello team!
I have checked the issues and there was a closed issue on supporting quic protocol
issue number : 4103
isn't there any plan to support that yet?

**Comment by flrgh** [comment]
Hello @hesamhamdarsi :wave: 

Kong is built atop of OpenResty, which itself is based on NGINX. Support of http/3 is on the [NGINX roadmap](https://www.nginx.com/blog/our-roadmap-quic-http-3-support-nginx/), but it is still under [active development](https://hg.nginx.org/nginx-quic/shortlog/quic) and [not production-ready](https://quic.nginx.org/readme.html). So at this point the most appropriate answer I can give is that Kong will likely support http/3 once the feature is mainlined in NGINX and makes it into OpenResty.

**Comment by WeetA34** [comment]
Hello @flrgh 
Fyi, it seems HTTP/3 is supported in OpenResty 1.25.3.1 released in January 2024.

**Reply by chronolaw** [reply]
Yes, OpenResty 1.25.3.1 supports HTTP/3, but it is experimental, not stable, we are still in discussion for it.

**Reply by epowell** [reply]
As of Oct 2024, it appears to be included in official packages by default now: https://openresty.org/en/ann-1027001001.html

How can we enable this? Thanks :)

**Reply by alice101-dev** [reply]
> Yes, OpenResty 1.25.3.1 supports HTTP/3, but it is experimental, not stable, we are still in discussion for it.

OpenResty natively supports HTTP/3 (QUIC) starting from version OpenResty 1.25.3.1
